dotCMS/src/main/java/com/dotcms/rendering/velocity/services/PageRenderUtil.java

```diff
@@ -514,4 +510,40 @@ private String getPersonaTagToIncludeContent(final HttpServletRequest request, f
 
         return hasPersonalizations ? currentPersonaTag : MultiTree.DOT_PERSONALIZATION_DEFAULT;
     }
+
+    private static class ContainerUUIDPersona {
+        Map<String, List<String>> containerUuidPersona     = Maps.newHashMap();
+
+        public void add(
+                final Container container,
+                final String uniqueUUIDForRender,
+                final PersonalizedContentlet personalizedContentlet) {
+
+            get(container, uniqueUUIDForRender, personalizedContentlet)
+                    .add(personalizedContentlet.getContentletId());
+        }
+
+        private List<String> get(Container container, String uniqueUUIDForRender, PersonalizedContentlet personalizedContentlet) {
+            return containerUuidPersona
+                    .computeIfAbsent(
+                            getKey(container, uniqueUUIDForRender, personalizedContentlet),
+                            k -> Lists.newArrayList()
+                    );
+        }
+
+        private String getKey(Container container, String uniqueUUIDForRender, PersonalizedContentlet personalizedContentlet) {
+            return container.getIdentifier() + uniqueUUIDForRender + personalizedContentlet.getPersonalization();
+        }
+
+        public long getSize(
+                final Container container,
+                final String uniqueUUIDForRender,
+                final PersonalizedContentlet personalizedContentlet) {
+            return get(container, uniqueUUIDForRender, personalizedContentlet).size();
+        }
+
+        public Iterable<? extends Map.Entry<String, List<String>>> entrySet() {
```

Comment:
should return a Set?

Reply:
done https://github.com/dotCMS/core/pull/19447/commits/bbd176303ed633ddea71b2af08cb868208201451#diff-135d6f9eb01eb89ef742b83753636596a41a9e3ab757abd4347b57e80ae892e0R548